### DISCUSSION TITLE
Custom InMemoryDataset not finding custom get() function

### DISCUSSION THREAD

**Discussion by s-n-naik** [pyg-team/pytorch_geometric#7914] [Q&A]
Hello,

When I generate a dataset for the first time, the custom processed() function is called and the custom get() is used when I index the resulting dataset using dataset[idx] and it returns two graphs (with diff augmentations) as I have coded.

However, when I reload the data from an already generated data.pt, the InMemoryDataset only returns the original graph and doesn't use my custom get() - it seems to just use the default one. 

How can I make sure that it uses my custom get() both times? Has anyone else seen this?

**Comment by rusty1s** [comment]
This sounds strange and definitely not intended. How can I reproduce this?
